### ISSUE TITLE
Do `no-native-reassign` and `no-redeclare` cover the same use case?

### ISSUE THREAD

**Issue by rioc0719** [eslint/eslint#4435]
When using `no-declare` with the `builtinGlobals` option set to `true`, it seems to cover the same use case as `no-native-reassign`. Is there a reason that there are two separate rules that both check for assignments to built-in objects?

By the way, I am using eslint 1.9.0, and the issue where it was decided to have both options is #3070.

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by nzakas** [comment]
They are not the same. This is a redeclare:

```js
var Object = foo;
```

This is reassign:

```js
Object = foo
```
